CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
+__category__ is a collection of api calls that are exposed to the client
+to do things inside that category. For example, in the storage category
+generally one wants to upload and download objects from storage so the
+apis exposed to the client will represent that functionality. Because
+Amplify is pluggable, a __plugin__ of your choosing will provide the
+actual implementation behind that api interface.  Using the same example
+of Storage, the plugin we choose might be `AWSS3StoragePlugin`, which
+would then implement each api call from the category with a service call
+or set of service calls to S3, the underlying storage provider of the
+AWS plugin.
+
+## Getting Started
+
+First, ensure that you have installed the latest stable version of Android
+Studio / the Android SDK.
+
+Configure your environment, so that the `ANDROID_HOME` and `JAVA_HOME`
+environment variables are set. A convenient way of doing this is to add them
+into `~/.bashrc`. On a Mac, the SDK and Java installation used by the SDK may
+be found:
+
+```
+export ANDROID_HOME=~/Library/Android/sdk
+export JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home
+```
+Note: JDK 11, 12, 13, etc. have known issues and are not supported.
+
+Now, clone the Amplify Android project from GitHub.
+
+```
+git clone git@github.com:aws-amplify/amplify-android.git
+```
+Load this project into Android Studio by selecting File > Open, and choosing
+the root directory of the project (`amplify-android`). Alternately, cd into this
+top-level directory.
+
+In Android Studio, build the project by clicking the Hammer icon, "Make Project
+⌘F9". If working on the command line, you can do the same thing via:
+
+```
+./gradlew build
+```
+
+## Tools
+[Gradle](https://gradle.org) is used for all [build and dependency management](https://developer.android.com/studio/build).
+
+Some widely used dependencies are:
+
+1. [org.json](https://developer.android.com/reference/org/json/JSONObject) is
+   baked into Android, and is used for all modeling with JSON.
+2. [Gson](https://github.com/google/gson/blob/master/README.md#gson) is used
+   for serialization and deserialization.
+3. [OkHttp](https://github.com/square/okhttp#okhttp) is used for network operations.
+
+_Unit and component tests_, which run on your development machine, use:
+
+1. [jUnit](https://github.com/junit-team/junit4/blob/r4.13/README.md#junit-4), to make assertions
+2. [Mockito](https://javadoc.io/static/org.mockito/mockito-core/1.10.19/org/mockito/Mockito.html#1), to mock dependencies
+3. [Robolectric](https://github.com/robolectric/robolectric/blob/master/README.md),
+   to simulate an Android device when unit tests execute on your machine.
+
+_Instrumentation tests_, which run on an Android device or emulator use
+AndroidX test core, runner, and a jUnit extension. See Android's notes on
+[using AndroidX for test](https://developer.android.com/training/testing/set-up-project).
+
+## Workflows
```

Comment:
Can we add something about how to get acclimated to the codebase? Mentioning the Good First Issue label for folks starting off, for instance.

Reply:
Later in this doc we have a `Finding contributions to work on` section. _However_, we should think of a way to beef it up. Currently, it says just:
> Looking at the existing issues is a great way to find something to work on. 

The `Adding code to support a new feature` sub-section contains a brief tour of the modules / code layout. This could be promoted to a general paragraph for the `Workflows`, or other section?